src/tools/Move.ts

```diff
@@ -0,0 +1,77 @@
+import { Tool, Tracker } from '@/tools/Tool'
+import Konva from 'konva'
+import { CanvasElement } from '@/types/Canvas'
+import throttle from 'lodash.throttle'
+
+export default class FreeDraw implements Tool {
+  // eslint-disable-next-line
+  private group: Konva.Group
+  constructor (public readonly name: string,
+               public temporary: boolean) {
+    this.group = new Konva.Group()
+  }
+
+  // eslint-disable-next-line
+  mouseDownAction = (e: Konva.KonvaPointerEvent, canvasElement: CanvasElement, layer: Konva.Layer, _socket: WebSocket): void => {
+    canvasElement.data = [e.evt.x, e.evt.y]
+    canvasElement.hasMoved = false
+    canvasElement.tracker = Tracker.MOVE
+    canvasElement.id = ''
+    canvasElement.tool = {
+      name: this.name,
+      temporary: this.temporary
+    }
+    const group = e.target.parent
+    if (group instanceof Konva.Group) {
+      this.group = group
+      canvasElement.id = this.group.attrs.id
+    }
+  }
+
+  // eslint-disable-next-line
+  mouseMoveAction = throttle((e: Konva.KonvaPointerEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket): void => {
+    if (!canvasElement.hasMoved) {
+      canvasElement.hasMoved = true
+    }
+    if (canvasElement.id !== '') {
```

Comment:
`canvasElement.id` is better, or are we just checking that it isn't an empty string. What about if it is null or undefined?

Reply:
Has to look for empty string